src/main/java/org/isf/admission/rest/AdmissionController.java

```diff
@@ -212,73 +211,68 @@ public ResponseEntity<List<AdmittedPatientDTO>> allAdmittedPatients()
+					throws OHServiceException {
 		LOGGER.info("Get admitted patients search terms: {}", searchTerms);
 		if (admissionRange != null && admissionRange.length == 2) {
 			LOGGER.debug("Get admissions started between {} and {}", admissionRange[0], admissionRange[1]);
 		}
 		if (dischargeRange != null && dischargeRange.length == 2) {
-			LOGGER.debug("Get admissions ended between {} and {}", dischargeRange[0], dischargeRange[1]);
+			LOGGER.debug("Get admissions that end between {} and {}", dischargeRange[0], dischargeRange[1]);
 		}
-		
+
 		List<AdmittedPatient> admittedPatients = admissionManager.getAdmittedPatients(admissionRange, dischargeRange, searchTerms);
 		if (admittedPatients.isEmpty()) {
 			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
 		}
-		
+
 		return ResponseEntity.ok(admittedMapper.map2DTOList(admittedPatients));
 	}
-	
+
 	/**
-	 * Get all the {@link Admission}s for the specified {@link Patient} code.
+	 * Get all the {@link Admission}s that start in the specified range
 	 * 
-	 * @param patientCode
+	 * @param admissionRange
 	 * @return the {@link List} of found {@link Admission} or NO_CONTENT otherwise.
 	 * @throws OHServiceException
 	 */
 	@GetMapping(value = "/admissions", produces = MediaType.APPLICATION_JSON_VALUE)
 	public ResponseEntity<List<AdmissionDTO>> getAdmissions(
-			@RequestParam(name = "patientcode", defaultValue = "0", required = false) int patientcode,
-			@RequestParam(name = "admissionrange", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] admissionRange,
-			@RequestParam(name = "dischargerange", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] dischargeRange)
-			throws OHServiceException {
-		LOGGER.info("Get admissions of patients by id: {}", patientcode);
-		if (admissionRange != null && admissionRange.length == 2) {
-			LOGGER.debug("Get admissions started between {} and {}", admissionRange[0], admissionRange[1]);
-		}
-		if (dischargeRange != null && dischargeRange.length == 2) {
-			LOGGER.debug("Get admissions ended between {} and {}", dischargeRange[0], dischargeRange[1]);
-		}
-		
-		List<AdmittedPatient> admittedPatients = new ArrayList<AdmittedPatient>();
-		if (patientcode == 0) {
-			admittedPatients = admissionManager.getAdmittedPatients(admissionRange, dischargeRange, "");
-		} else {
-			admittedPatients = admissionManager.getAdmittedPatients(admissionRange, dischargeRange, Integer.toString(patientcode));
-		}
-		if (admittedPatients.isEmpty()) {
+					@RequestParam(name = "admissionrange") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] admissionRange)
+					throws OHServiceException {
+		LOGGER.debug("Get admissions started between {} and {}", admissionRange[0], admissionRange[1]);
+
+		List<Admission> admissions = admissionManager.getAdmissions(admissionRange[0], admissionRange[1], 0, 0);
```

Comment:
What about page infos ?  Why are you setting page and size to 0 ?

Reply:
It could be OK but, it could be better to return pagination info to the user(currentPage, hasNext, hastPrevious, totalCount, numberOf elements). That we, return object will be something a  "{  pageInfo: {...}, data: [...] }" What do you think ?

Reply:
It would be nice, but we didn't do that for GET /patients (see below)

Should we fix parameters now and tackle a different return object across the application in a dedicated issue? I think it is a general change...

```
	@GetMapping(value = "/patients", produces = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity<List<PatientDTO>> getPatients(
			@RequestParam(value="page", required=false, defaultValue="0") Integer page,
			@RequestParam(value="size", required=false, defaultValue=DEFAULT_PAGE_SIZE) Integer size) throws OHServiceException {
		LOGGER.info("Get patients page: {}  size: {}", page, size);
		List<Patient> patients = patientManager.getPatient(page, size);
		List<PatientDTO> patientDTOS = patients.stream().map(pat -> {
			return patientMapper.map2DTO(pat);
		}).collect(Collectors.toList());
        if (patientDTOS.isEmpty()){
            return ResponseEntity.status(HttpStatus.NO_CONTENT).body(patientDTOS);
        }
		return ResponseEntity.ok(patientDTOS);
	}
```

Reply:
Alright! can you create a dedicated, well described, issue?
So for this PR I will just add parameters, ok?

Reply:
OK, let me do that !